File: chegaJa_back/src/main/java/com/tech/chegaJa/domain/model/FormaPagamento.java
package com.tech.chegaJa.domain.model;

import com.tech.chegaJa.domain.dto.FormaPagamentoDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.time.LocalDate;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class FormaPagamento {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String numeroCartao;
    private String bandeira;
    private LocalDate datavencimento;
    private String codigo;
    private String nomeTitular;
    private String tipoPagamento;

    public FormaPagamentoDto toDto() {
        return new FormaPagamentoDto(id,numeroCartao,bandeira,datavencimento,codigo,nomeTitular,tipoPagamento);
    }
}

File: chegaja_frontend/lib/models/forma_pagamento.dart
class FormaPagamentoForm {
  String? numeroCartao;
  String? bandeira;
  String? dataVencimento;
  String? codigo;
  String? nomeTitular;
  String? tipoPagamento;

  FormaPagamentoForm(
      {this.numeroCartao,
      this.bandeira,
      this.dataVencimento,
      this.codigo,
      this.nomeTitular,
      this.tipoPagamento});

  FormaPagamentoForm.fromJson(Map<String, dynamic> json) {
    numeroCartao = json['numeroCartao'];
    bandeira = json['bandeira'];
    dataVencimento = json['dataVencimento'];
    codigo = json['codigo'];
    nomeTitular = json['nomeTitular'];
    tipoPagamento = json['tipoPagamento'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['numeroCartao'] = this.numeroCartao;
    data['bandeira'] = this.bandeira;
    data['dataVencimento'] = this.dataVencimento;
    data['codigo'] = this.codigo;
    data['nomeTitular'] = this.nomeTitular;
    data['tipoPagamento'] = this.tipoPagamento;
    return data;
  }
}
